lib/web_ui/lib/src/engine/canvas_pool.dart

```diff
@@ -0,0 +1,761 @@
+      contextHandle.reset();
+      --_saveContextCount;
+    }
+  }
+
+  @override
+  void translate(double dx, double dy) {
+    super.translate(dx, dy);
+    if (_canvas != null) {
+      context.translate(dx, dy);
+    }
+  }
+
+  @override
+  void scale(double sx, double sy) {
+    super.scale(sx, sy);
+    if (_canvas != null) {
+      context.scale(sx, sy);
+    }
+  }
+
+  @override
+  void rotate(double radians) {
+    super.rotate(radians);
+    if (_canvas != null) {
+      context.rotate(radians);
+    }
+  }
+
+  @override
+  void skew(double sx, double sy) {
+    super.skew(sx, sy);
+    if (_canvas != null) {
+      context.transform(1, sy, sx, 1, 0, 0);
+      //                |  |   |   |  |  |
+      //                |  |   |   |  |  f - vertical translation
+      //                |  |   |   |  e - horizontal translation
+      //                |  |   |   d - vertical scaling
+      //                |  |   c - horizontal skewing
+      //                |  b - vertical skewing
+      //                a - horizontal scaling
+      //
+      // Source: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/transform
+    }
+  }
+
+  @override
+  void transform(Float64List matrix4) {
+    super.transform(matrix4);
+    // Canvas2D transform API:
+    //
+    // ctx.transform(a, b, c, d, e, f);
+    //
+    // In 3x3 matrix form assuming vector representation of (x, y, 1):
+    //
+    // a c e
+    // b d f
+    // 0 0 1
+    //
+    // This translates to 4x4 matrix with vector representation of (x, y, z, 1)
+    // as:
+    //
+    // a c 0 e
+    // b d 0 f
+    // 0 0 1 0
+    // 0 0 0 1
+    //
+    // This matrix is sufficient to represent 2D rotates, translates, scales,
+    // and skews.
+    if (_canvas != null) {
+      context.transform(matrix4[0], matrix4[1], matrix4[4], matrix4[5],
+          matrix4[12], matrix4[13]);
+    }
+  }
+
+  void clipRect(ui.Rect rect) {
+    super.clipRect(rect);
+    if (_canvas != null) {
+      _clipRect(context, rect);
+    }
+  }
+
+  void _clipRect(html.CanvasRenderingContext2D ctx, ui.Rect rect) {
+    ctx.beginPath();
+    ctx.rect(rect.left, rect.top, rect.width, rect.height);
+    ctx.clip();
+  }
+
+  void clipRRect(ui.RRect rrect) {
+    super.clipRRect(rrect);
+    if (_canvas != null) {
+      _clipRRect(context, rrect);
+    }
+  }
+
+  void _clipRRect(html.CanvasRenderingContext2D ctx, ui.RRect rrect) {
+    final ui.Path path = ui.Path()..addRRect(rrect);
+    _runPath(ctx, path);
+    ctx.clip();
+  }
+
+  void clipPath(ui.Path path) {
+    super.clipPath(path);
+    if (_canvas != null) {
+      html.CanvasRenderingContext2D ctx = context;
+      _runPath(ctx, path);
+      ctx.clip();
+    }
+  }
+
+  void drawColor(ui.Color color, ui.BlendMode blendMode) {
+    html.CanvasRenderingContext2D ctx = context;
+    contextHandle.blendMode = blendMode;
+    contextHandle.fillStyle = color.toCssString();
+    contextHandle.strokeStyle = '';
```

Comment:
Still missing how a `filter` applied in a `drawPaint` won't leak into `drawColor` 🤔 

Example:

```dart
c.drawPaint(... pass Paint with filter...);
// at this point, because we didn't reset the previous paint the `filter` is still there.
c.drawColor(... some color and blend mode ...);
```

Reply:
Done. Added filter = 'none';